### PR TITLE
feat(Alert): wrap button under text + remove truncation on smaller screensizes

### PR DESCRIPTION
**Pull request by dreamwasp**
### Overview

Wraps the alert button below alert text  and does not truncate Alerts when on the smallest screensize. Also added font-smoothing to match Figma alerts 

image

### PR Checklist

- [x] Related to designs: [Designs](https://www.figma.com/file/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?type=design&node-id=30186-40358&mode=design&t=ljOUYqcXytX1bDlv-0)
- [x] Related to JIRA ticket: WEB-2520
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change
- [x] This PR includes testing instructions tests for the code change
- [x] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

1. Go to the Alert page
2. Alerts should looks identical to how they do now.
3.  Set your screensize to mobile
4. See the truncation be removed and main buttons wrapped to under the text 

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/36837)  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/4412)  | [Portal Env](http://www.google.fr/ 'Named link title')   |

### CODE REVIEW

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6570c6bf8916b006318ac28a--gamut-preview.netlify.app>https://6570c6bf8916b006318ac28a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6570c6bf8916b006318ac28a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.10.8-alpha.033203.0<br/>@codecademy/gamut-kit@0.6.372-alpha.033203.0<br/>@codecademy/styleguide@66.8.8-alpha.033203.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.10.8-alpha.e19f19.0<br/>@codecademy/gamut-kit@0.6.372-alpha.e19f19.0<br/>@codecademy/styleguide@66.8.8-alpha.e19f19.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6579ca2828cbcb26533fa140--gamut-preview.netlify.app>https://6579ca2828cbcb26533fa140--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6579ca2828cbcb26533fa140>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.10.8-alpha.3bbc6c.0<br/>@codecademy/gamut-kit@0.6.372-alpha.3bbc6c.0<br/>@codecademy/styleguide@66.8.8-alpha.3bbc6c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6579d35256e1312d1608f0d9--gamut-preview.netlify.app>https://6579d35256e1312d1608f0d9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6579d35256e1312d1608f0d9>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.10.9-alpha.1978e2.0<br/>@codecademy/gamut-kit@0.6.373-alpha.1978e2.0<br/>@codecademy/styleguide@66.8.9-alpha.1978e2.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://657c7ed9201fdd0738681c04--gamut-preview.netlify.app>https://657c7ed9201fdd0738681c04--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/657c7ed9201fdd0738681c04>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.10.9-alpha.94cc2c.0<br/>@codecademy/gamut-kit@0.6.373-alpha.94cc2c.0<br/>@codecademy/styleguide@66.8.9-alpha.94cc2c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://657c83e68b2c2805bd44dd13--gamut-preview.netlify.app>https://657c83e68b2c2805bd44dd13--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/657c83e68b2c2805bd44dd13>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.10.9-alpha.22cae5.0<br/>@codecademy/gamut-kit@0.6.373-alpha.22cae5.0<br/>@codecademy/styleguide@66.8.9-alpha.22cae5.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://658074201d85b8637e439792--gamut-preview.netlify.app>https://658074201d85b8637e439792--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/658074201d85b8637e439792>Deploy Logs</a></p>

**Review by aresnik11 — COMMENTED** [review]
i'm not seeing the alert button below alert text when the truncated alert is expanded

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6596d311b4ad2a009bbb65a8--gamut-preview.netlify.app>https://6596d311b4ad2a009bbb65a8--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6596d311b4ad2a009bbb65a8>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.10.9-alpha.fdb040.0<br/>@codecademy/gamut-kit@0.6.373-alpha.fdb040.0<br/>@codecademy/styleguide@66.8.9-alpha.fdb040.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65983cd31b14f031cfb8dc00--gamut-preview.netlify.app>https://65983cd31b14f031cfb8dc00--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65983cd31b14f031cfb8dc00>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.10.12-alpha.31a485.0<br/>@codecademy/gamut-kit@0.6.376-alpha.31a485.0<br/>@codecademy/styleguide@66.9.2-alpha.31a485.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65a69ffc9d5c1b043672fab5--gamut-preview.netlify.app>https://65a69ffc9d5c1b043672fab5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65a69ffc9d5c1b043672fab5>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.10.12-alpha.e3bc06.0<br/>@codecademy/gamut-kit@0.6.376-alpha.e3bc06.0<br/>@codecademy/styleguide@66.9.2-alpha.e3bc06.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.10.12-alpha.449bab.0<br/>@codecademy/gamut-kit@0.6.376-alpha.449bab.0<br/>@codecademy/styleguide@66.9.2-alpha.449bab.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65a6d2c7df78f30e8e82658d--gamut-preview.netlify.app>https://65a6d2c7df78f30e8e82658d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65a6d2c7df78f30e8e82658d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.10.12-alpha.26532a.0<br/>@codecademy/gamut-kit@0.6.376-alpha.26532a.0<br/>@codecademy/styleguide@66.9.2-alpha.26532a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65a6f926df78f329e1826910--gamut-preview.netlify.app>https://65a6f926df78f329e1826910--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65a6f926df78f329e1826910>Deploy Logs</a></p>

File: packages/gamut/src/Alert/Alert.tsx [hunk 1]
```diff
@@ -55,13 +62,38 @@ export const Alert: React.FC<AlertProps> = ({
...
+
+  const gridTemplateColumns = useMemo(() => {
+    return !isDesktop
+      ? `max-content minmax(0, 1fr) repeat(1, max-content)`
+      : getGridTemplateColumns({
+          cta: ctaExists,
+          onClose: !!onClose,
+          truncated,
+        });
+  }, [ctaExists, isDesktop, onClose, truncated]);
```
Comment (dreamwasp) [inline packages/gamut/src/Alert/Alert.tsx:91]: these are all being calculated based on screensize so i want to make sure they're not being calculated too often

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.10.12-alpha.a2328b.0<br/>@codecademy/gamut-kit@0.6.376-alpha.a2328b.0<br/>@codecademy/styleguide@66.9.2-alpha.a2328b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65a83852d049df07dad09b91--gamut-preview.netlify.app>https://65a83852d049df07dad09b91--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65a83852d049df07dad09b91>Deploy Logs</a></p>

File: packages/gamut/src/Alert/__mocks__/index.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export const mockUseMedia = jest.fn();
```
Comment (aresnik11) [inline packages/gamut/src/Alert/__mocks__/index.ts:1]: if this isnt being used anywhere else can it just be an inline mock in the alert test?

Comment (dreamwasp) [reply packages/gamut/src/Alert/__mocks__/index.ts]: i was getting errors at some point for declaring it within the file, seems to be working now!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.10.12-alpha.76c526.0<br/>@codecademy/gamut-kit@0.6.376-alpha.76c526.0<br/>@codecademy/styleguide@66.9.2-alpha.76c526.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65aec370a99ce44033649df1--gamut-preview.netlify.app>https://65aec370a99ce44033649df1--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65aec370a99ce44033649df1>Deploy Logs</a></p>

File: packages/gamut/src/Alert/__mocks__/index.ts (hunk 2 above)
Comment (dreamwasp) [reply packages/gamut/src/Alert/__mocks__/index.ts]: nvm, this error is still happening in ci 😞

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65aedc7d9b48905bce93576d--gamut-preview.netlify.app>https://65aedc7d9b48905bce93576d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65aedc7d9b48905bce93576d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.10.12-alpha.3a1c11.0<br/>@codecademy/gamut-kit@0.6.376-alpha.3a1c11.0<br/>@codecademy/styleguide@66.9.2-alpha.3a1c11.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.10.12-alpha.b4225e.0<br/>@codecademy/gamut-kit@0.6.376-alpha.b4225e.0<br/>@codecademy/styleguide@66.9.2-alpha.b4225e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65aee00bba900457bab4f4a3--gamut-preview.netlify.app>https://65aee00bba900457bab4f4a3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65aee00bba900457bab4f4a3>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65afd8a3bfc74023d653fb0c--gamut-preview.netlify.app>https://65afd8a3bfc74023d653fb0c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65afd8a3bfc74023d653fb0c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.10.12-alpha.496130.0<br/>@codecademy/gamut-kit@0.6.376-alpha.496130.0<br/>@codecademy/styleguide@66.9.2-alpha.496130.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.10.12-alpha.f8eed9.0<br/>@codecademy/gamut-kit@0.6.376-alpha.f8eed9.0<br/>@codecademy/styleguide@66.9.2-alpha.f8eed9.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65afdb965f398426dc9ff789--gamut-preview.netlify.app>https://65afdb965f398426dc9ff789--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65afdb965f398426dc9ff789>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.10.12-alpha.70e94a.0<br/>@codecademy/gamut-kit@0.6.376-alpha.70e94a.0<br/>@codecademy/styleguide@66.9.2-alpha.70e94a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65afef9237dee20a7abf134c--gamut-preview.netlify.app>https://65afef9237dee20a7abf134c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65afef9237dee20a7abf134c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.10.12-alpha.175b04.0<br/>@codecademy/gamut-kit@0.6.376-alpha.175b04.0<br/>@codecademy/styleguide@66.9.2-alpha.175b04.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65b02465b0258e360fac8f28--gamut-preview.netlify.app>https://65b02465b0258e360fac8f28--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65b02465b0258e360fac8f28>Deploy Logs</a></p>

File: .vscode/settings.json [hunk 3]
```diff
@@ -28,7 +28,7 @@
     "editor.wordWrap": "wordWrapColumn",
     "editor.wordWrapColumn": 120,
     "editor.codeActionsOnSave": {
-      "source.fixAll": true
+      "source.fixAll": "explicit"
```
Comment (aresnik11) [inline .vscode/settings.json:31]: what does this do?

Comment (dreamwasp) [reply .vscode/settings.json:31]: this is unintentional, idk where it's coming from but its happening to me and kenny on and off

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65b14fd8cbb0c028f45603c8--gamut-preview.netlify.app>https://65b14fd8cbb0c028f45603c8--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65b14fd8cbb0c028f45603c8>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.10.12-alpha.f6c30f.0<br/>@codecademy/gamut-kit@0.6.376-alpha.f6c30f.0<br/>@codecademy/styleguide@66.9.2-alpha.f6c30f.0<br/></pre>

**Review by aresnik11 — APPROVED** [review]
LGTM!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.10.12-alpha.adcbd1.0<br/>@codecademy/gamut-kit@0.6.376-alpha.adcbd1.0<br/>@codecademy/styleguide@66.9.2-alpha.adcbd1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65b160120afe752d3cc78a6e--gamut-preview.netlify.app>https://65b160120afe752d3cc78a6e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65b160120afe752d3cc78a6e>Deploy Logs</a></p>

**Pull request merged**